x-pack/plugins/cloud_security_posture/public/pages/rules/rules_flyout.tsx

```diff
@@ -48,9 +52,21 @@ const tabs = [
 
 type RuleTab = typeof tabs[number]['id'];
 
-export const RuleFlyout = ({ onClose, rule }: RuleFlyoutProps) => {
+export const RuleFlyout = ({ onClose, rule, refetchStatus }: RuleFlyoutProps) => {
   const [tab, setTab] = useState<RuleTab>('overview');
+  const postRequestChangeRulesStatus = useChangeCspRuleStatus();
 
+  const switchRuleStatus = async () => {
+    const rulesObjectRequest = {
+      benchmark_id: rule?.metadata.benchmark.id,
+      benchmark_version: rule?.metadata.benchmark.version,
+      rule_number: rule?.metadata.benchmark.rule_number,
+      rule_id: rule?.metadata.id,
+    };
+    const nextRuleStatus = rule?.status === 'muted' ? 'unmute' : 'mute';
+    await postRequestChangeRulesStatus(nextRuleStatus, [rulesObjectRequest]);
+    await refetchStatus();
```

Comment:
consider renaming

```suggestion
    await refetchRulesStatus();
```

the original name made me think its the status of the refetch